### PR TITLE
Add support for Dependency Graph Snapshots endpoint

### PR DESCRIPTION
**Pull request by be0x74a**
Starts Dependency Graph service support by starting with the [Create Snapshot](https://docs.github.com/en/rest/dependency-graph/dependency-submission?apiVersion=2022-11-28#create-a-snapshot-of-dependencies-for-a-repository) endpoint

### CODE REVIEW

File: README.md
Comment (gmlewis) [inline README.md:398]: Please do not modify version information in PRs... you can revert the changes to this file.
Versions are maintained and controlled by the repo owners.  Thanks.

File: github/github.go
Comment (gmlewis) [inline github/github.go:31]: Please revert this change.

File: github/dependency_graph_snapshots.go
Comment (gmlewis) [inline github/dependency_graph_snapshots.go:33]: Plese revert these changes.  In this repo, everything lives at the top-level namespace of the "github" module.  As a result, generic names like this (e.g. `SUCCESS`) are simply too general and therefore confusing to the user of this repo, since they are specifically only used for the DependencyGraph service.

Instead, we document using GoDoc-style comments what the values of certain fields can be (according to the official GitHub v3 API documentation).

Additionally, all your structs have names that are too general and need to be more specific.
Please take a browse through the code base of this repo and see how structs relating to specific endpoints and service have been names, and follow likewise.

I'm going to stop my review at this point so that you can please make the appropriate modifications.  Thanks.

**Review by gmlewis — CHANGES_REQUESTED** [review]

File: github/dependency_graph_snapshots.go
Comment (be0x74a) [reply github/dependency_graph_snapshots.go:33]: Alright, thanks for the feedback, will address the comments

Comment (be0x74a) [reply github/dependency_graph_snapshots.go]: Hey @gmlewis, just addressed the comments. Thanks for the feedback!
Hopefully the comments around the possible values for the string structs are what you're expecting

File: github/dependency_graph_snapshots.go
Comment (gmlewis) [inline github/dependency_graph_snapshots.go:33]: Please let's not do this.

Please delete the 3 new types with extremely long names, and make those fields below simply of type `*string` and move the comments into the structs below, immediately preceeding each respective field.

File: github/dependency_graph_snapshots.go
Comment (gmlewis) [inline github/dependency_graph_snapshots.go:35]: Each new struct type needs a Godoc-style comment that starts with its name and make a complete statement about its purpose, ending with a period, so that the auto-generated documentation reads nicely.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @be0x74a !

**Comment by gmlewis** [comment]
@be0x74a - we'll leave this PR open for a week or two, but if we haven't heard back from you, we will close this PR as abandoned.

File: github/dependency_graph_snapshots_test.go
Comment (Al8900) [inline github/dependency_graph_snapshots_test.go:24]: image

**Review by Al8900 — APPROVED** [review]

**Comment by be0x74a** [comment]
> @be0x74a - we'll leave this PR open for a week or two, but if we haven't heard back from you, we will close this PR as abandoned.

@gmlewis finally got to it! Sorry for the delay and the force-push, was trying to appease codecov and make it use a recent base but I completely failed...

File: github/dependency_graph_snapshots.go
Comment (gmlewis) [inline github/dependency_graph_snapshots.go:82]:
```suggestion
	ID        int64        `json:"id"`
```

File: github/dependency_graph_snapshots.go
Comment (gmlewis) [inline github/dependency_graph_snapshots.go:84]:
```suggestion
	CreatedAt *Timestamp `json:"created_at,omitempty"`
	Message   *string    `json:"message,omitempty"`
```

File: github/dependency_graph_snapshots.go
Comment (gmlewis) [inline github/dependency_graph_snapshots.go:90]:
```suggestion
	Result *string `json:"result,omitempty"`
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @be0x74a !
This looks much better. Just a few minor tweaks, please, then we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

**Comment by be0x74a** [comment]
Thanks for the input @gmlewis!

**Review by gmlewis — APPROVED** [review]
Thank you, @be0x74a !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by tomfeigin — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @tomfeigin !
Merging.

**Pull request merged**
